kroxylicious-app/pom.xml

```diff
@@ -46,17 +47,17 @@
         <!-- project dependencies - runtime and compile -->
         <dependency>
             <groupId>io.kroxylicious</groupId>
-            <artifactId>kroxylicious-api</artifactId>
+            <artifactId>kroxylicious-filter-api</artifactId>
```

Comment:
We had separate api/filter-api modules previously and squashed them back together as it was complicating life and we couldn't work out when consumers would only want the filter api and not the "runtime" API... 

Part of me does wonder about shifting the generated code to its own module such as `kroxylicious-kafka-36-api` and potentially `kroxylicious-kafka-40-api` but I'm not at all sure about the merits or implied complexity of that.

Reply:
There's the question of what it should be _called_ and what it should _contain_, and they're different things. I think we made the right decision to combine those previous modules. But calling it `kroxylicious-api` implies it's the _only_ API, and that's unlikely to remain true for long. In the other PR we have `kroxylicous-kms`, which could legitimately be called `kroxylicous-kms-api`. In the future we might grow kubernetes specific functionlity and want a `kroxylicious-kubernetes-api`. We might end up wanting to have a type safe Java API for generating config (similar to what we're doing already for test purposes), and that might be `kroxylicious-config-api`. These APIs have different uses and different audiences.

The point about the Kafka dependency deserves more thought, but it clearly not for this PR.

Reply:
I get the motivation for `kroxylicious-filter-api` and agree is should be done.  However, I'm not sure if now is the moment to do it.   It is `io.kroxylicious.proxy.plugin.Plugin` that gives me pause.  I know filters are the only users of this API, but  they are really generic code.  One answer would be `kroxylicious-plugin-api`, but that might be too granular.  

I would defer renaming kroxylicious-api -> kroxylicious-filter-api.

Reply:
> I know filters are the only users of this API

This is already not true: The in-memory KMS module doesn't actually implement any filter but depends on `kroxylicious-(filter-)?api` just for `@Plugin`.

I agree that we should probably factor out a `-plugin-api` module. But I don't want to waste too much time thinking about this right now. So I've reverted this rename in the interests of getting this PR merged.